trigger-actions-framework/main/default/classes/FinalizerHandler.cls

```diff
@@ -126,12 +126,38 @@ public with sharing virtual class FinalizerHandler {
 		Context context
 	) {
 		Object dynamicInstance;
+		String namespace = finalizerMetadata.Apex_Class_Namespace__c;
 		String className = finalizerMetadata.Apex_Class_Name__c;
 		if (FinalizerHandler.isBypassed(className)) {
 			return;
 		}
+
 		try {
-			dynamicInstance = Type.forName(className).newInstance();
+			System.Type actionType = Type.forName(namespace, className);
+			/** Type.forName(fullyQualifiedName) allowed some messyness and ambiguity in dealing with namespace
+			 *   If config does not provide the correct namespace (likely if upgrading from older versions of this framework) we need to fallback in two scenarios
+			 *     - package and class namespaced but namespace wasn't specified
+			 *     - namespace is actually in the class field in the form namespace.classname
+			 */
+			// try shared Namespace
+			if( actionType == null ) {
+				// Get the namespace of the current class.
+				String[] parts = String.valueOf(MetadataTriggerHandler.class).split('\\.', 2);
```

Comment:
This looks a bit odd to me. `MetadataTriggerHandler` class will always have no namespace in the unlocked package.

Is this intended for when someone uses the Trigger Actions Framework within their own Managed package?

Reply:
Or namespace unlocked package, but yes, that is exactly the intent here as that is how we are using the framework internally (in Namepsaced packages)